rfcs/0166-nix-formatting.md

```diff
@@ -0,0 +1,1140 @@
+---
```

Comment:
Imo a Nix code formatter should never change the AST or the evaluation result (ignoring impurities like `builtins.unsafeGetAttrPos`). If at all, it could be an opt-in feature in the future. I think this should definitely be out of scope of this RFC.

Reply:
Currently, any strings containing bash are Nixpkgs specific and thus out of scope of this RFC. A first step towards that direction would be to have syntactic annotation of the language or "format" of the string's content, so that the formatter can work with it in the first place. But then, the other restrictions still apply.